libs/blocks/global-navigation/blocks/navDropdown/dropdown.js

```diff
@@ -0,0 +1,179 @@
+import { getAnalyticsValue, toFragment, decorateCta } from '../../utilities/utilities.js';
+import { localizeLink } from '../../../../utils/utils.js';
+
+const decorateHeadline = (elem) => {
+  if (!(elem instanceof HTMLElement)) return null;
+
+  // TODO: proper handling of aria-expanded across devices
+  const headline = toFragment`<div
+    class="feds-popup-headline"
+    role="heading"
+    aria-level="2"
+    aria-haspopup="true"
+    aria-expanded="false">
+    ${elem.textContent.trim()}
+  </div>`;
+
+  // TODO: move proper logic to accessibility,
+  // this is just for demo functionality purposes
+  headline.addEventListener('click', (e) => {
+    e.preventDefault();
+
+    const openPopup = document.querySelector('.feds-popup-headline[aria-expanded = "true"]');
+
+    if (openPopup && openPopup !== headline) {
+      openPopup.setAttribute('aria-expanded', 'false');
+    }
+
+    const currentState = headline.getAttribute('aria-expanded');
+
+    if (currentState === 'false') {
+      headline.setAttribute('aria-expanded', 'true');
+    } else {
+      headline.setAttribute('aria-expanded', 'false');
+    }
+  });
+
+  // Since heading is turned into a div, it can be safely removed
+  elem.remove();
+
+  return headline;
+};
+
+const decorateLinkGroup = (elem, index) => {
+  if (!(elem instanceof HTMLElement)) return null;
```

Comment:
Let's add a safeguard for `const link = elem.querySelector('a');` here and discard the link group if it doesn't have a link. Wonder if we also have any debugging logs we could add?
```suggestion
  if (!(elem instanceof HTMLElement) || !elem.querySelector('a')) return null;
```

Reply:
Btw on this note, I'll create a ticket that we have to add some lana logs - I want to observe any errors we see after launching.

Reply:
Yup, good point with checking that the group has a link, will accept your change 👍